### PR TITLE
style(Core/Achievement): Improve error msg for achievement_reward

### PR DESCRIPTION
**Pull request by BarbzYHOOL**
## CHANGES PROPOSED:
- Change a outError to outDebug
- Fix typos and improve the texts to waste less time for the user
- Added some slightly useful comments for future devs  

## ISSUES ADDRESSED:
- Closes 

## TESTS PERFORMED:
- Can't test at the moment, but it's mostly "cosmetic" changes

## HOW TO TEST THE CHANGES:

- Not much to test really, better review the code directly
- Insert various rows in the achievement_reward table with missing values (cf http://www.azerothcore.org/wiki/achievement_reward note: I updated the wiki too) and see the console messages when typing `.reload achievement_reward`. For the outDebug, you'll need to enable debug in worldserver.conf

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 1]
```diff
@@ -2782,35 +2782,35 @@ void AchievementGlobalMgr::LoadRewards()
...
-        reward.titleId[1]   = fields[2].GetUInt32();
+        reward.titleId[0]   = fields[1].GetUInt32(); // Alliance title
+        reward.titleId[1]   = fields[2].GetUInt32(); // Horde title
         reward.itemId       = fields[3].GetUInt32();
-        reward.sender       = fields[4].GetUInt32();
+        reward.sender       = fields[4].GetUInt32(); // The sender of the mail (a creature from creature_template)
         reward.subject      = fields[5].GetString();
-        reward.text         = fields[6].GetString();
+        reward.text         = fields[6].GetString(); // Body in DB
         reward.mailTemplate = fields[7].GetUInt32();
```
Comment (BarbzYHOOL) [inline src/server/game/Achievements/AchievementMgr.cpp:2796]: i'm not sure this method of getting fields by number is good at all :P

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 2]
```diff
@@ -2782,35 +2782,35 @@ void AchievementGlobalMgr::LoadRewards()
...
-            sLog->outErrorDb("Table `achievement_reward` has wrong achievement (Entry: %u), ignore", entry);
+            sLog->outErrorDb("Table `achievement_reward` has wrong achievement (Entry: %u). Ignoring.", entry);
             continue;
         }
 
         AchievementReward reward;
-        reward.titleId[0]   = fields[1].GetUInt32();
-        reward.titleId[1]   = fields[2].GetUInt32();
+        reward.titleId[0]   = fields[1].GetUInt32(); // Alliance title
+        reward.titleId[1]   = fields[2].GetUInt32(); // Horde title
```
Comment (BarbzYHOOL) [inline src/server/game/Achievements/AchievementMgr.cpp:2791]: I wanted to change these vars directly but they're used in the whole file, so too risky to change it all for me

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 3]
```diff
@@ -2820,51 +2820,52 @@ void AchievementGlobalMgr::LoadRewards()
...
+                sLog->outErrorDb("Table `achievement_reward` (Entry: %u) has mailTemplate (%u) and mail Subject/Body. To use the column mailTemplate, Subject and Body must be empty.", entry, reward.mailTemplate);
         }
 
         if (reward.itemId)
         {
             if (!sObjectMgr->GetItemTemplate(reward.itemId))
             {
-                sLog->outErrorDb("Table `achievement_reward` (Entry: %u) has invalid item id %u, reward mail will not contain item.", entry, reward.itemId);
+                // Not sure it's an error, it's probably an outDebug instead, because we can simply send a mail with no reward, right?
+                sLog->outErrorDb("Table `achievement_reward` (Entry: %u) has invalid item_template id %u. Reward mail will not contain any item.", entry, reward.itemId);
```
Comment (BarbzYHOOL) [inline src/server/game/Achievements/AchievementMgr.cpp:2868]: here if someone confirms, we can put an outDebug

**Comment by Helias** [comment]
I have no suggestions for this, but I think that we don't need test for this because it's just an improvements of some texts

**Comment by BarbzYHOOL** [comment]
yeah me too
the only thing that could be tested is to insert one row with only one title reward like "titleH" and try to display it in debug mode but seriously, seems like a waste of testing time :D

Note: I put outDebug here https://github.com/azerothcore/azerothcore-wotlk/pull/2852/files#diff-5b6d79cde3a43bd23e1c8c8366e6ada3R2806 but itcould be outString too, I just felt like it was not wanted and would make the user worried for no reason

**Comment by Kitzunu** [comment]
ty for the PR!

**Pull request merged**